pkg/input/input.go

```diff
@@ -39,3 +40,31 @@ func Password(promptText string) (string, error) {
 	fmt.Print("\n")
 	return string(bytePassword), nil
 }
+
+// Structure to hold content required for displayig prompts required for promptui library functions
+type PromptContent struct {
+	Label string
+}
+
+type PromptRunner interface {
+	Run() (int, string, error)
+}
+
+func GetYesNoSelector(pc PromptContent) *promptui.Select {
+	return &promptui.Select{
+		Label: pc.Label,
+		Items: []string{"y", "n"},
+	}
+}
+
+// Gets a y/n confirmation from the user for the given prompt content using the promptui library and returns a boolean accordingly
+func PromptGetConfirmation(runner PromptRunner) (bool, error) {
```

Comment:
we should be adding unit tests for these functions since it has been added to `pkg` which is a shared package.

Reply:
@neel-astro would appreciate you help here, quite unsure how to add unit tests for these components.

Reply:
```Golang
func TestPromptGetConfirmation(t *testing.T) {
	runner := GetYesNoSelector(PromptContent{Label: "test label, enter y/n"})
	runner.Keys = &promptui.SelectKeys{Next: promptui.Key{Code: rune('S')}, Prev: promptui.Key{Code: rune('W')}, PageUp: promptui.Key{Code: rune('D')}, PageDown: promptui.Key{Code: rune('A')}}
	tests := []struct {
		name        string
		inputString string
		want        bool
		wantErr     bool
	}{
		{
			name:        "basic yes case",
			inputString: "\n",
			want:        true,
			wantErr:     false,
		},
		{
			name:        "basic no case",
			inputString: "S\n",
			want:        false,
			wantErr:     false,
		},
		{
			name:        "no input case",
			inputString: "",
			want:        false,
			wantErr:     true,
		},
	}
	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			runner.Stdin = io.NopCloser(strings.NewReader(tt.inputString))
			got, err := PromptGetConfirmation(runner)
			if (err != nil) != tt.wantErr {
				t.Errorf("PromptGetConfirmation() error = %v, wantErr %v", err, tt.wantErr)
				return
			}
			if got != tt.want {
				t.Errorf("PromptGetConfirmation() = %v, want %v", got, tt.want)
			}
		})
	}
}
```

Reply:
Added in b3bd5e2